### DISCUSSION TITLE
How to jump to folder with higher ranked subfolders?

### DISCUSSION THREAD

**Discussion by tommyalatalo** [ajeetdsouza/zoxide#764] [Q&A] [answered]
I'm not sure if I'm missing something in how zoxide is supposed to be used, as I'm a previous autojump user, but I'm having some issues getting to where I want to go, even after using zoxide for some time.

An example; if I want to jump to my `/home/user/git` folder, I would expect to do it by `z git`, as I have no other folder trees that end with `/git`. Instead I end up in other folders because they're ranked higher in the zoxide database, listing them for this example with `z git -l` gives me these top ranked folders:
```
96095      /home/user/git/esphome
113258     /home/user/git/homelab/nomad/jobs
1344805    /home/user/git/homelab/nomad
4597161    /home/user/git
5087465    /home/user/git/homelab/vault
```
When I do `z git` with the above ranking in the database, I end up in `~/git/homelab/vault`, which is not what I wanted.

Is it possible to configure zoxide to prioritize the last subfolder as the match, or am I simply doing something wrong?
How can I jump to the `~/git` folder when other subfolders under it are ranked higher?

**Comment by ajeetdsouza** [comment, marked as answer]
`z git` is only supposed to match directories (not subdirectories). Also, `z git -l` is not a valid zoxide command.

Are you sure you're using zoxide?

**Reply by tommyalatalo** [reply]
Okay, wow, I was completely blindsided by this, you're right, I didn't realize that `z` in my shell is not zoxide for some reason:
```
❯ which z
z: aliased to zshz 2>&1
```
I'm not sure where that alias is being set to be honest, but zshz is the zsh builtin z, yes?
I have `eval "$(zoxide init zsh --cmd j)"` as the very last line in my `.zshrc`, shouldn't that override z as well?

edit: I had an alias `alias j=z`, after removing this things seem to work properly as obviously zoxide is now called correctly and not the zsh builtin. Thanks for the input :)

**Reply by ajeetdsouza** [reply]
zsh doesn't have any builtin for `z`, you're likely using a plugin called `zsh-z`.

**Reply by tommyalatalo** [reply]
Thanks, yes you were right, I had zsh-z installed, it's been so long since I set up my shell environment that I forgot that it was there. I now have zoxide working exactly as expected and it's awesome, I really wanted to start using `zi/ji` and it's been great.
